ebean-datasource/src/main/java/io/ebean/datasource/pool/PooledConnection.java

```diff
@@ -448,6 +448,11 @@ public void close() throws SQLException {
     if (status == STATUS_IDLE) {
       throw new SQLException(IDLE_CONNECTION_ACCESSED_ERROR + "close()");
     }
+    boolean mayHaveUncommittedChanges = !autoCommit && !readOnly && status == STATUS_ACTIVE;
```

Comment:
Note: We assume, that every connection with `autoCommit = false` and `readOnly = false` may have uncommitted changes if it is not committed or rolled back.

Other CP (like hikari) puts a wrapper around everything (down to [ResultSet.updateRow](https://github.com/brettwooldridge/HikariCP/blob/dev/src/main/java/com/zaxxer/hikari/pool/ProxyResultSet.java#L69)) to detect, if the connection may be dirty.

And then comes DB2... which starts a new UOW on `getSchema()`

Reply:
This looks all good.